### PR TITLE
Merge-devel-v1.0.4

### PR DESCRIPTION
**Pull request by marek-szydlowski-wizzdev**

### CODE REVIEW

File: MicroPython/scripts/generate_terraform.py [hunk 1]
```diff
@@ -29,10 +33,28 @@ def save_terraform_output_as_file(_output_file_name: str):
         outfile.write(tf_output_str)
 
 
+def _check_terraform():
+    """Checks if terraform binary exists in the system"""
+
+    terraform_path = shutil.which(TERRAFORM_BINARY_NAME)
+    if terraform_path is not None:
```
Comment (marcin-nawrocki-wizzdev) [inline MicroPython/scripts/generate_terraform.py:40]: In the scenario, where user have both local terraform binary, and terraform installed in system, which one will be called?

Comment (marek-szydlowski-wizzdev) [reply MicroPython/scripts/generate_terraform.py:40]: Terraform installed in the system

**Review by marcin-nawrocki-wizzdev — APPROVED** [review]
All good!

**Comment by marcin-nawrocki-wizzdev** [comment]
Changes:
Support for Windows; Improved Readme; Multiple fixes

**Review by przemyslaw-aszkowski-wizzdev — APPROVED** [review]

**Pull request merged**